### PR TITLE
O3-2720: Visit stop time input not working in edit mode

### PR DESCRIPTION
**Pull request by sherrif10**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Other

Ticket :  https://openmrs.atlassian.net/browse/O3-2720

In the original code `visitStartDate` and `visitStopDate` are used to initialize the start and stop dates, respectively. If visitStartDate or visitStopDate is provided, they are used; otherwise, the current date is used. however the time-related fields visitStartTime, visitStartTimeFormat, visitStopTime, visitStopTimeFormat) are always initialized based on the provided or current date. in the fix i ensured consistency for both start and stop date. the code uses newDate() to get current date and time. Let me know your suggestions.  cc @chibongho @rbuisson @vasharma05 

https://github.com/openmrs/openmrs-esm-patient-chart/assets/48015736/81df5554-0ad7-41dc-90d4-04c1cc061001

After
Screenshot from 2024-01-25 22-46-20

### CODE REVIEW

**Comment by chibongho** [comment]
I don't think this fixes the underlying problem. This change seems to change the default start and end time in the form to current date / time. We still want to keep it defaulted to the original start / end time of the visit. The problem is with the `minDate` and `maxDate` calculation passed to the `<DatePicker>`

**Comment by sherrif10** [comment]
Hello @chibongho , i updated the code accordingly .

**Comment by sherrif10** [comment]
Hi @denniskigen .Looking forward to hear from you.

**Comment by denniskigen** [comment]
Hey, @sherrif10, thanks for working on this. Is the first video you linked to in the PR description meant to replicate the problem? I cannot tell from viewing it what the problem is.

**Comment by sherrif10** [comment]
Hello @denniskigen , Yes the first vedio is meant to replicate this issue.

File: packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx [hunk 1]
```diff
@@ -497,8 +497,9 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
     return [maxVisitStartDatetime, minVisitStopDatetime];
   }, [visitToEdit]);
 
-  const visitStartDate = getValues('visitStartDate');
-  minVisitStopDatetime = minVisitStopDatetime ?? Date.parse(visitStartDate.toLocaleString());
+  const visitStartDate = getValues('visitStartDate') || new Date();
```
Comment (denniskigen) [inline packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx:500]: Why are we optionally calling `new Date()` here? It looks like its already getting called conditionally higher up when initializing `defaultValues` [higher up](https://github.com/openmrs/openmrs-esm-patient-chart/blob/ac5f146ac02036214ad81240acc179817799aac4/packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx#L141).

**Comment by sherrif10** [comment]
You are right , this pattern ensures that if visitToEdit doesn't have a start or stop datetime, the default value is set to the current date.

File: packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx [hunk 2]
```diff
@@ -497,8 +497,10 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
     return [maxVisitStartDatetime, minVisitStopDatetime];
   }, [visitToEdit]);
 
-  const visitStartDate = getValues('visitStartDate');
-  minVisitStopDatetime = minVisitStopDatetime ?? Date.parse(visitStartDate.toLocaleString());
+  const visitStartDateValue = getValues('visitStartDate');
+  const visitStartDate = visitStartDateValue !== undefined ? visitStartDateValue : new Date();
+  const minVisitStopDatetimeFallback = Date.parse(visitStartDate.toLocaleString());
+  minVisitStopDatetime = minVisitStopDatetime || minVisitStopDatetimeFallback;
```
Comment (vasharma05) [inline packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx:503]:
```suggestion
  const visitStartDate = getValues('visitStartDate') ?? new Date();
  minVisitStopDatetime = minVisitStopDatetime ?? Date.parse(visitStartDate.toLocaleString());
```

Isn't this the whole change you did?

Comment (sherrif10) [reply packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx:503]: hI @vasharma05 ,Here is an updated screen shot 
Screenshot from 2024-01-29 14-25-18

Comment (sherrif10) [reply packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx]: Hi @denniskigen @vasharma05 . i addressed your comments. let me know your suggestions.

**Review by vasharma05 — APPROVED** [review]

**Pull request merged**